Source/WebCore/rendering/RenderBlock.cpp

```diff
@@ -334,19 +334,26 @@ void RenderBlock::styleWillChange(StyleDifference diff, const RenderStyle& newSt
     RenderBox::styleWillChange(diff, newStyle);
 }
 
-static bool borderOrPaddingLogicalWidthChanged(const RenderStyle& oldStyle, const RenderStyle& newStyle)
+bool RenderBlock::scrollbarWidthDidChange(const RenderStyle& oldStyle, const RenderStyle& newStyle)
+{
+    return (includeVerticalScrollbarSize() || includeHorizontalScrollbarSize()) && oldStyle.scrollbarWidth() != newStyle.scrollbarWidth();
```

Comment:
`borderOrPaddingLogicalWidthChanged` is careful to test changes separately on each axis, yet this function looks at both axes, so I suggest passing an argument that tells you whether to check just for horizontal or vertical scrollbar.